src/Server/HTTP/HTTPServerResponse.cpp

```diff
@@ -117,7 +118,11 @@ void HTTPServerResponse::sendBuffer(const void * buffer, std::size_t length)
     hs.flush();
 
     if (request && request->getMethod() != HTTPRequest::HTTP_HEAD)
-        WriteBufferFromPocoSocket(session.socket(), write_event).write(static_cast<const char *>(buffer), length);
+    {
+        auto wb = WriteBufferFromPocoSocket(session.socket(), write_event);
```

Comment:
It can be `AutoFinalizedWriteBuffer` as well, but doesn't really matter I guess

Reply:
I do not want to use `AutoFinalizedWriteBuffer` widely.

Reply:
In that case we want the exception from the `finalize` to be thrown at error.